object_database/web/cells.py

```diff
@@ -917,11 +920,9 @@ def __init__(self, inner):
         super().__init__()
 
         self.children = {"____contents__": Cell.makeCell(inner)}
-        self.contents = """
-        <div class='card-title'>
-          ____contents__
-        </div>
-        """
+        self.contents = HTMLElement.div().add_child(
+            HTMLTextContent("____contents__")
```

Comment:
so one of the advantages of your approach is that instead of having this ridiculous ____X__ pattern, we could have an additional html element called 'CellsChild' that is supposed to get swapped out for an appropriate cells child later once its properly built. This could really clean up the code because the references to the child items would be much more obvious.

Probably not for this PR, though.

Reply:
Yeah I think there are a few different ways we could merge the functionality together a bit more, whether it's as you've suggested or if every cell has it's own `self.element` that is the main wrapper element for the cell, or if the `Cell` class itself requires a `print_on` (or `printOn`) in similar fashion for any string generation functions. We should discuss it for sure.

Reply:
We could also have the init  or some other chosen method return `self` and have more dom-js style patterns such as:
```python
HTMElement.div(
      some stuff).p(
           some more stuff)....
```

this would be an anti-pattern of sorts but potentially worth considering given that we are trying to find some balance between the python world and the dom-html-js world. 

We can also add unique ids, and some "DOM/Node" class, that all the HTMLElements inherit, that you can do placeholder by-id replacements. For example:

```python
element = HTMLElement('div', id=1).with_children(
    HTMLElement('img', is_self_closing=True, id=2).add_class("primary-img"),
    HTMLElement('p', id=3).add_class("main-typography").add_child(
        HTMLTextContent("This is a placeholder", id=4)
    )
)

element.replaceByID(id=4, html=HtmlTextContent("We just changed this element"))
```

which would be equivalent to 
```python
element = HTMLElement('div', id=1).with_children(
    HTMLElement('img', is_self_closing=True, id=2).add_class("primary-img"),
    HTMLElement('p', id=3).add_class("main-typography").add_child(
        HTMLTextContent("We just changed this element", id=4)
    )
)
```

these ids could be user defined (semantic or int), auto-generated as default etc 

so a few things to discuss.

Reply:
So I was merely thinkiing that right now we write '____contents__' inline in the html, and separately have a dictionary with {'____contents__': some_cell} and this is how we register the placement in the DOM of our child cells - it's more explicit and wouldn't require direct text substitution.